code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/extensions/RemoteViewsExtensions.kt

```diff
@@ -148,28 +147,28 @@ internal fun RemoteViews.setNotificationBodyTextColor(
 }
 
 /**
- * Sets the large icon for the provided [RemoteViews]. If a large icon contains a filename
- * only then the large icon is set from a bundle image resource. If a large icon contains a URL,
- * the large icon is downloaded then set.
+ * Sets the image for the provided [RemoteViews]. If a image contains a filename
+ * only then the image is set from a bundle image resource. If the image contains a URL,
+ * the image is downloaded then set.
  *
- * @param largeIcon `String` containing the large icon to use
+ * @param image `String` containing the image to use
+ *
+ * @return `Boolean` true if the image was set, false otherwise
  */
-internal fun RemoteViews.setRemoteViewLargeIcon(largeIcon: String?) {
-    if (largeIcon.isNullOrEmpty()) {
+internal fun RemoteViews.setRemoteViewImage(
+    image: String?,
+    containerViewId: Int
+): Boolean {
+    if (image.isNullOrEmpty()) {
         Log.trace(
             PushTemplateConstants.LOG_TAG,
             SELF_TAG,
-            "Null or empty large icon string found, large icon will not be applied."
+            "Null or empty image string found, image will not be applied."
         )
-        setViewVisibility(R.id.large_icon, View.GONE)
-        return
-    }
-
-    if (UrlUtils.isValidUrl(largeIcon)) {
-        setRemoteLargeIcon(largeIcon)
-    } else {
-        setBundledLargeIcon(largeIcon)
+        setViewVisibility(containerViewId, View.GONE)
+        return false
     }
+    return setRemoteImage(image, containerViewId) || setBundledImage(image, containerViewId)
```

Comment:
Seems like the order of assignment matters. Can you add a note here for clarity.